workspace/python/ds-pack/v1/projector-proxy/jupyter_projector_proxy/__init__.py

```diff
@@ -0,0 +1,41 @@
+import os
+import shutil
+
+SYS_PROJECTOR_CONFIG_DIR = "/usr/local/share/projector"
+USER_PROJECTOR_CONFIG_DIR = os.path.join(os.environ["HOME"], ".projector")
+
+
+def setup_projector():
+    _copy_projector_config()
+    return {
+        "command": ["projector", "run", "pycharm"],
+        "launcher_entry": {
+            "title": "PyCharm",
+            "icon_path": os.path.join(
+                os.path.dirname(os.path.abspath(__file__)),
+                "icons",
+                "pycharm_icon.svg",
+            ),
+        },
+        # default projector config runs on port 9999
+        "port": 9999,
```

Comment:
Does this work with multiple users at the same time.  We previously had a bug where a user would hog the port.  @conor-james-sheehan didn't you introduce changes to fix that?

Reply:
Nope - started but never found a solution, dropped it after a while